src/utils/sourcemaps.rs

```diff
@@ -578,19 +627,68 @@ impl SourceMapProcessor {
         }
     }
 
+    /// Old versions of Sentry cannot assemble artifact bundles straight away, they require
+    /// that those bundles are associated to a release.
+    ///
+    /// This function checks if given chunk upload options about the server's capabilities
+    /// and only creates a release if the server requires that.
+    fn initialize_legacy_release_upload(&self, context: &UploadContext) -> Result<()> {
```

Comment:
This logic is a bit convoluted and I'm not super happy with it, but it should ensure that all of this works for old and new sentries.